### PR TITLE
Add support for `color` attribute to radio

### PR DESCRIPTION
**Pull request by rtrompier**
I have added an Input COLOR to allow tu user to choose the color like the checkbox component.
The color can be `primary`, `accent`, or `warn`.

screen shot 2017-04-11 at 21 40 13
screen shot 2017-04-11 at 21 40 23
screen shot 2017-04-11 at 21 40 30

### CODE REVIEW

File: src/lib/radio/_radio-theme.scss [hunk 1]
```diff
@@ -12,28 +12,64 @@
...
+      }
+
+      .mat-radio-inner-circle {
+        background-color: mat-color($warn);
+      }
+
+      .mat-radio-ripple .mat-ripple-element {
+        background-color: mat-color($warn, 0.26);
+      }
+    }
```
Comment (DevVersion) [inline src/lib/radio/_radio-theme.scss]: I have the feeling that we could move that into a mixin? So we don't have to repeat code three times.

Comment (rtrompier) [inline src/lib/radio/_radio-theme.scss]: Yes sure ;) 
I will create a mixin, you're right.

**Comment by rtrompier** [comment]
@DevVersion Added mixin & Default color (accent)

**Comment by jelbourn** [comment]
@rtrompier could you rebase this PR?

**Comment by rtrompier** [comment]
Should be good ;)

File: src/lib/radio/_radio-theme.scss [hunk 2]
```diff
@@ -9,33 +9,57 @@
   $background: map-get($theme, background);
   $foreground: map-get($theme, foreground);
 
-  .mat-radio-outer-circle {
-    border-color: mat-color($foreground, secondary-text);
-  }
+    .mat-radio-button {
+        //default color (accent)
```
Comment (jelbourn) [inline src/lib/radio/_radio-theme.scss]: Our indentation is +2, not +4

File: src/lib/radio/_radio-theme.scss [hunk 3]
```diff
@@ -9,33 +9,57 @@
...
+        &:not(.mat-primary):not(.mat-accent):not(.mat-warn) {
```
Comment (jelbourn) [inline src/lib/radio/_radio-theme.scss]: I don't think this `:not` expression should be necessary; the radio should just have `accent` be the default color.

File: src/lib/radio/_radio-theme.scss [hunk 4]
```diff
@@ -9,33 +9,57 @@
...
+        &.mat-warn {
+            @include radio-theme-color($warn, $foreground);
+        }
+    }
+}
 
-  .mat-radio-ripple .mat-ripple-element {
-    background-color: mat-color($accent, 0.26);
-  }
+@mixin radio-theme-color($color, $foreground) {
```
Comment (jelbourn) [inline src/lib/radio/_radio-theme.scss]: This mixin is going to generate a lot of css output; only styles that are directly affected by the palette swap should be defined within.

File: src/lib/radio/radio.ts [hunk 5]
```diff
@@ -396,6 +406,9 @@ export class MdRadioButton implements OnInit, AfterViewInit, OnDestroy {
   /** Whether the ripple effect on click should be disabled. */
   private _disableRipple: boolean;
 
+  /** Current color. Can be `primary`, `accent`, or `warn`. */
+  private _color: string;
```
Comment (jelbourn) [inline src/lib/radio/radio.ts]: The type should use the string literal values
```ts
private _color: 'primary' | 'accent' | 'warn';
```

**Comment by DevVersion** [comment]
@rtrompier Thanks for your work on this PR. It looks like your PR is no longer up-to-date and there isn't any activity here.

I'm going to address the color binding for radios in #5068

File: src/lib/radio/radio.ts (hunk 5 above)
Comment (Polyterative) [reply src/lib/radio/radio.ts]: Is there any way to use straight color values like for example `#00FF00` or `rgb(255,100,0)` from HTML ?

**Pull request closed without merging**